src/excalidraw-backend/types/socket.data.ts

```diff
@@ -25,9 +26,24 @@ export type SocketData = {
   /***
    * If the user can update the content of the whiteboard
    */
-  update: boolean;
+  collaborator: boolean;
   /***
    * The session of the user connected with the socket
    */
   session?: Session;
+  /**
+   * Amount of times this socket has failed saving
+   */
+  failedSaves: number;
+  /**
+   * Can the socket save the whiteboard.</br>
+   * This is individual of the <i>collaborator</i> flag.</br> A socket can still save without being a collaborator.
```

Comment:
I don't think we should allow a socket that has update privilege but didn't get into edit mode because of licensing reasons save the wb for others even if he's watching all the edits. Is this true? `A socket can still save without being a collaborator`

Reply:
why not?

Reply:
clear now after our call